src/Adapter/Address/CommandHandler/EditCartAddressHandler.php

```diff
@@ -155,7 +156,11 @@ private function createEditAddressCommand(EditCartAddressCommand $cartCommand, C
             $addressCommand->setAddress2($cartCommand->getAddress2());
         }
         if (null !== $cartCommand->getStateId()) {
-            $addressCommand->setStateId($cartCommand->getStateId()->getValue());
+            if($cartCommand->getStateId() === NoStateId::NO_STATE_ID_VALUE) {
```

Comment:
Once you change back the setter with int parameter this shouldn't be needed anymore

Reply:
copy that too !